### PR TITLE
Fix clock offset scaling, especially in sp3 interpolation

### PR DESCRIPTION
**Pull request by gwbres**
- [x] Solves #219

### CODE REVIEW

**Comment by lnicola** [comment]
Ha! You can merge without a review! :grinning:

**Comment by gwbres** [comment]
> Ha! You can merge without a review! 😀

Yeah, sorry but I was pretty sure about this one, and we had a very interesting pending discussion (see the Issue portal) that I wanted to update

**Comment by lnicola** [comment]
It's fine, just that a while ago you couldn't merge without a review and couldn't self-approve either.

File: sp3/src/tests/parser_3d.rs [hunk 1]
```diff
@@ -82,15 +82,15 @@ mod test {
         for (epoch, sv, clock) in sp3.sv_clock() {
             assert_eq!(epoch, Epoch::from_str("2019-10-27T00:00:00 GPST").unwrap());
             if sv == sv!("C01") {
-                assert_eq!(clock, 63.035497, "bad clock data");
+                assert!((clock - 63.035497E-6).abs() < 1E-9, "bad clock data");
```
Comment (lnicola) [inline sp3/src/tests/parser_3d.rs:85]: Not sure it's worth using it, but https://docs.rs/approx.

Comment (gwbres) [reply sp3/src/tests/parser_3d.rs:85]: Yeah I should look into adding this (only in test mode )

**Pull request merged**